File: backend/src/main/java/com/erp/backend_erp/dto/cattleSales/CattleSaleDto.java
package com.erp.backend_erp.dto.cattleSales;

import java.math.BigDecimal;
import java.util.List;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class CattleSaleDto {
    
    private Long id;
    private String tipoVenta;
    private String fechaVenta;
    private String horaEmision;
    private String numeroFactura;
    private Double precioKilo;
    private Double pesoTotal;
    private BigDecimal subtotal;
    private BigDecimal iva;
    private BigDecimal descuentos;
    private BigDecimal total;
    private String moneda;
    private String formaPago;
    private String destino;
    private String observaciones;
    private Long farmId;
    private Long terceroId;
    private List<Long> cattleIds;
    private List<CreateCattleSaleItemDto> items;
    
}

File: frontend/src/app/modules/cattle-sales/domain/dto/CattleSaleDto.ts
import { CreateCattleSaleItemDto } from './create-cattle-sale-item.dto';

export interface CattleSaleDto {
  id: number;
  numeroFactura: string;
  fechaVenta: string; // formato yyyy-MM-dd
  horaEmision: string; // formato HH:mm
  moneda: string;
  formaPago: string;
  terceroId: number;
  tipoIdentificacion: string;
  numeroIdentificacion: string;
  nombreRazonSocial: string;
  direccion: string;
  telefono: string;

  destino: string;
  observaciones: string;

  precioKilo: number;
  pesoTotal: number;
  subtotal: number;
  iva: number;
  descuentos: number;
  precioTotal: number;

  items: CreateCattleSaleItemDto[];
}
